### DISCUSSION TITLE
Instantsearch with other engine

### DISCUSSION THREAD

**Discussion by laitauchocolat** [algolia/instantsearch#4825] [Q&A]
Instantsearch is open source, but can I use it with other engine like...elasticsearch... easily. I dont want to create a backend that map response from engine A to algolia or to completely rewrite algoli client. Anyone knows ?

**Comment by francoischalifour** [comment]
We have a guide that explains [how to use InstantSearch with a custom backend](https://www.algolia.com/doc/guides/building-search-ui/going-further/backend-search/in-depth/backend-instantsearch/js/). You can then plug any other search solution with InstantSearch, including Elasticsearch. You only need to map the Algolia search parameters into the Elasticsearch parameters, which we do not provide.

**Comment by britisharmy** [comment]
> Instantsearch is open source, but can I use it with other engine like...elasticsearch... easily. I dont want to create a backend that map response from engine A to algolia or to completely rewrite algoli client. Anyone knows ?

There are numerous ways to do that. If you have data in elastic engine, then you can use this package for Vue or React, https://github.com/appbaseio/reactivesearch   

That being said, i prefer algolia since its easier to use and progressive.
